lib/python_version_utils.pm

```diff
@@ -77,8 +81,37 @@ sub remove_installed_pythons() {
         my $python_versions = script_output("rpm -q $python3_spec_release | awk -F \'-\' \'{print \$2}\'");
         record_info("Python version", "$python_versions:$default_python");
         next if ($python_versions eq $default_python);
-        zypper_call("remove $python3_spec_release-base");
+        if (is_transactional) {
+            trup_call("pkg rm $python3_spec_release-base");
+            process_reboot(expected_grub => 1, trigger => 1);
+        } else {
+            zypper_call("rm $python3_spec_release-base");
+        }
+    }
+}
+
+sub get_available_fullstack_pythons {
+    # Check if python-rpm-macros is already installed
+    my $is_installed = script_run("rpm -qa | grep -q 'python-rpm-macros'");
+
+    # If the package is not installed, proceed with the installation
+    if ($is_installed) {
```

Comment:
This reads backwars - `if is_installed, then install` (the code is likely right, as script run will have returned 0 on 'installed' and 1 on uninstalled; but that makes the variable name wrong.